lib/internal/test_runner/snapshot.js

```diff
@@ -77,7 +77,7 @@ class SnapshotFile {
   }
 
   setSnapshot(id, value) {
-    this.snapshots[templateEscape(id)] = value;
+    this.snapshots[keyEscape(id)] = value;
```

Comment:
Since this appears to duplicate much of `templateEscape()`, couldn't we do this instead: `keyEscape(templateEscape(id))`

Reply:
I tried it but then it starts failing as we over-escape things, the problem is with the `\\` to `\\\\` replace